runtime/js/30_fs.js

```diff
@@ -148,48 +154,101 @@
   }
 
   async function remove(
     path,
     options = {},
   ) {
-    await core.opAsync("op_remove_async", {
-      path: pathFromURL(path),
-      recursive: !!options.recursive,
-    });
+    await core.opAsync(
+      "op_remove_async",
+      pathFromURL(path),
+      !!options.recursive,
+    );
   }
 
   function renameSync(oldpath, newpath) {
-    ops.op_rename_sync({
-      oldpath: pathFromURL(oldpath),
-      newpath: pathFromURL(newpath),
-    });
+    ops.op_rename_sync(
+      pathFromURL(oldpath),
+      pathFromURL(newpath),
+    );
   }
 
   async function rename(oldpath, newpath) {
-    await core.opAsync("op_rename_async", {
-      oldpath: pathFromURL(oldpath),
-      newpath: pathFromURL(newpath),
-    });
+    await core.opAsync(
+      "op_rename_async",
+      pathFromURL(oldpath),
+      pathFromURL(newpath),
+    );
   }
 
+  function createByteStruct(types) {
+    // types can be "date", "bool" or "u64".
+    // `?` prefix means optional on windows.
+    let offset = 0;
+    let str =
+      'const unix = Deno.build.os === "darwin" || Deno.build.os === "linux"; return {';
+    for (let [name, type] of ObjectEntries(types)) {
+      const optional = type.startsWith("?");
+      if (optional) type = type.slice(1);
+
+      if (type == "u64") {
+        if (!optional) {
+          str += `${name}: view[${offset}] + view[${offset + 1}] * 2**32,`;
+        } else {
+          str += `${name}: (unix ? (view[${offset}] + view[${
+            offset + 1
+          }] * 2**32) : (view[${offset}] + view[${
+            offset + 1
+          }] * 2**32) || null),`;
+        }
+      } else if (type == "date") {
+        str +=
+          `${name}: view[${offset}] === 0 ? null : new Date(view[${offset}] + view[${
+            offset + 1
+          }] * 2**32),`;
+      } else {
+        str += `${name}: !!(view[${offset}] + view[${offset + 1}] * 2**32),`;
+      }
+      offset += 2;
+    }
+    str += "};";
+    return [new Function("view", str), new Uint32Array(offset)];
+  }
+
+  const [statStruct, statBuf] = createByteStruct({
+    isFile: "bool",
+    isDirectory: "bool",
+    isSymlink: "bool",
+    size: "u64",
+    mtime: "date",
+    atime: "date",
+    birthtime: "date",
+    dev: "?u64",
+    ino: "?u64",
+    mode: "?u64",
+    nlink: "?u64",
+    uid: "?u64",
+    gid: "?u64",
+    rdev: "?u64",
+    blksize: "?u64",
+    blocks: "?u64",
+  });
+
   function parseFileInfo(response) {
     const unix = build.os === "darwin" || build.os === "linux";
     return {
       isFile: response.isFile,
       isDirectory: response.isDirectory,
       isSymlink: response.isSymlink,
       size: response.size,
-      mtime: response.mtime != null ? new Date(response.mtime) : null,
-      atime: response.atime != null ? new Date(response.atime) : null,
-      birthtime: response.birthtime != null
-        ? new Date(response.birthtime)
-        : null,
+      mtime: response.mtime != 0 ? new Date(response.mtime) : null,
```

Comment:
But what if mtime is 1/1/1970 at 00:00? 😉

Reply:
Fixed

Reply:
You updated the comparison for mtime, but not for atime and birthtime. Also you probably want triple equals

Reply:
oops fixed